graylog2-web-interface/src/components/sidecars/configuration-forms/EditConfigurationVariableModal.jsx

```diff
@@ -0,0 +1,146 @@
+import PropTypes from 'prop-types';
+import lodash from 'lodash';
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import { BootstrapModalForm, Input } from 'components/bootstrap';
+import CombinedProvider from 'injection/CombinedProvider';
+import ConfigurationHelperStyle from './ConfigurationHelper.css';
+
+const { ConfigurationVariableActions } = CombinedProvider.get('ConfigurationVariable');
+
+class EditConfigurationVariableModal extends React.Component {
+  static propTypes = {
+    create: PropTypes.boolean,
+    id: PropTypes.string,
+    name: PropTypes.string,
+    description: PropTypes.string,
+    content: PropTypes.string,
+    saveConfigurationVariable: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    create: false,
+    id: '',
+    name: '',
+    description: '',
+    content: '',
+  };
+
+  state = {
+    id: this.props.id,
+    name: this.props.name,
+    description: this.props.description,
+    content: this.props.content,
+    errors: {},
+  };
+
+  _hasErrors = () => {
+    return !lodash.isEmpty(this.state.errors);
+  };
+
+  openModal = () => {
+    this.modal.open();
+  };
+
+  _getId = (prefixIdName) => {
+    return prefixIdName + this.state.name;
```

Comment:
Could we use the ID instead of name to append to all IDs in the component? The name can change and with it the IDs inside the component, which may not be ideal.

As the ID will not be set when creating a new variable, we could use `this.state.id || 'new'`.